test/unit/nested.cpp

```diff
@@ -0,0 +1,31 @@
+#include "gtest/gtest.h"
+
+#include "RAJA/pattern/nested.hpp"
+
+#include <iostream>
+
+TEST(Nested, Basic)
+{
+  using namespace RAJA::nested;
+  using Index_type = RAJA::Index_type;
+  using pol = Policy<For<1, RAJA::seq_exec>, For<0, RAJA::seq_exec>>;
+  RAJA::nested::forall(pol{},
+                       RAJA::util::make_tuple(RAJA::RangeSegment(0, 5),
+                                              RAJA::RangeSegment(0, 5)),
+                       [=](Index_type i, Index_type j) {
+                         std::cout << i << " " << j << std::endl;
+                       });
+}
+
+TEST(Nested, collapse)
+{
+  using namespace RAJA::nested;
+  using Index_type = RAJA::Index_type;
+  using pol = Policy<Collapse<RAJA::seq_exec, CFor<1>, CFor<0>>>;
```

Comment:
I'm not sure what the best thing is to do for that name.  As of now, it does stand for "CollapseFor" but I'm trying to figure out if we can get away with just having one kind of For and apply it to both Collapse and the top level Policy to avoid having to have it be different.  Lacking that, I went with CFor to keep it terse for the first cut, but having it be more specific wouldn't be a terrible thing.